Tribler/Policies/BoostingManager.py

```diff
@@ -0,0 +1,493 @@
+        if not boost_source:
+            self._logger.info("Dropping torrent insert from removed source: %s", repr(torrent))
+            return
+        elif boost_source.archive:
+            torrent['preload'] = True
+            torrent['prio'] = 100
+
+        # If duplicates exist, set is_duplicate to True, except for the one with the most seeders.
+        duplicates = [other for other in self.torrents.values() if compare_torrents(torrent, other)]
+        if duplicates:
+            duplicates += [torrent]
+            healthiest_torrent = max([(torrent['num_seeders'], torrent) for torrent in duplicates])[1]
+            for duplicate in duplicates:
+                is_duplicate = healthiest_torrent != duplicate
+                duplicate['is_duplicate'] = is_duplicate
+                if is_duplicate and duplicate.get('download', None):
+                    self.stop_download(duplicate)
+
+        self.torrents[infohash] = torrent
+
+    def on_torrent_notify(self, subject, change_type, infohash):
+        """
+        Notify us when we have new seeder/leecher value in torrent from tracker
+        """
+        if infohash not in self.torrents:
+            return
+
+        self._logger.debug("infohash %s %s %s updated", subject, change_type, hexlify(infohash))
+
+        tdict = self.torrent_db.getTorrent(infohash, keys=['C.torrent_id', 'infohash', 'name',
+                                                           'length', 'category', 'status', 'num_seeders',
+                                                           'num_leechers'])
+
+        if tdict:
+            infohash_str = hexlify(tdict['infohash'])
+
+            new_seed = tdict['num_seeders']
+            new_leecher = tdict['num_leechers']
+
+            if new_seed - self.torrents[tdict['infohash']]['num_seeders'] \
+                    or new_leecher - self.torrents[tdict['infohash']]['num_leechers']:
+                self.torrents[tdict['infohash']]['num_seeders'] = new_seed
+                self.torrents[tdict['infohash']]['num_leechers'] = new_leecher
+                self._logger.info("infohash %s : seeder/leecher changed seed:%d leech:%d",
+                                  infohash_str, new_seed, new_leecher)
+
+    def scrape_trackers(self):
+        """
+        Manually scrape tracker by requesting to tracker manager
+        """
+
+        for infohash in list(self.torrents):
+            # torrent handle
+            lt_torrent = self.session.lm.ltmgr.get_session().find_torrent(lt.big_number(infohash))
+
+            peer_list = []
+            for i in lt_torrent.get_peer_info():
+                peer = LibtorrentDownloadImpl.create_peerlist_data(i)
+                peer_list.append(peer)
+
+            num_seed, num_leech = utilities.translate_peers_into_health(peer_list)
+            self._logger.debug("Seeder/leecher data translated from peers : seeder %s, leecher %s", num_seed, num_leech)
```

Comment:
Are all these lines of code executed to calculate two variables that are logged? Maybe that's a bit overkill? :)